app/src/main/java/com/helio/app/MainActivity.java

```diff
@@ -58,24 +71,176 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
     private void fetchState() {
         motors = new HashMap<>();
         HubClient client = new HubClient("http://10.0.2.2:4310/");
-        RegisterMotorRequest registerMotorRequest = new RegisterMotorRequest(
-                42,
+        testMotor(client);
+        testSchedule(client);
+        testLightSensors(client);
+        testMotionSensors(client);
+    }
+
+    private void testMotor(HubClient client) {
+        MotorSettingsRequest motorSettingsRequest = new MotorSettingsRequest(
                 "bedroom",
                 "1.2.3.4",
-                false
-        );
-        RenameMotorRequest renameMotorRequest = new RenameMotorRequest(42, "kitchen");
-        MoveMotorRequest moveMotorRequest = new MoveMotorRequest(42, 25);
+                false,
+                0, 0, 0, "style");
+        // use Handler to force the actions to happen in order
+        Handler handler = new Handler();
+
+        postTime += DELAY;
+        handler.postDelayed(() -> client.getAllMotors(motors), postTime);
+
+        postTime += DELAY;
+        handler.postDelayed(() -> client.addMotor(motors, motorSettingsRequest), postTime);
+
+        // This is extremely bad
+        AtomicInteger id = new AtomicInteger(-1);
+
+        postTime += DELAY;
+        handler.postDelayed(() -> id.set(new ArrayList<>(motors.keySet()).get(0)), postTime);
```

Comment:
The API was changed so that when you register something you do not specify an ID, instead the hub chooses the ID and sends it back. The problem is that for the following tests, we need to know the ID, that line gets the ID of the motor that has been sent back. It crashes if the hub isn't running, since it doesn't get anything back and there is no motor. This won't be a problem in the actual working version since you will only be able to update or move a motor once it already exists.

Reply:
Makes sense.
Increasing the delay removed the crashes for me on line 99 and 155, but it still happens on line 206. I don't think it's an issue given that these are just dummy actions.